### DISCUSSION TITLE
Coreapi topic

### DISCUSSION THREAD

**Discussion by pfk-beta** [encode/django-rest-framework#8602] [Question & Answer]
Hello,

Thanks for DRF software and greate community around it. Recently I came back to using DRF, and now my scenario is to create api service and client library. To not reinventing the wheel, I dediced to use coreapi, as it is demonstrated in documentation page (https://www.django-rest-framework.org/topics/api-clients/). So I play with coreapi cli, to get listings, created examples in python, created wrapper to play with my objects. Then I realize that I need to implement one of core usecases, creating object via API. It is simple, just `client.action(['api', 'create_object'], params={'name': 'test_name'})`. What a suprise!? It doesn't work:
```
coreapi.exceptions.ErrorMessage: <Error: 400 Bad Request>
    name: [
        "This field is required."
    ]
```
Yes, I'm doing it wrong, params are url parameters, you need post data. Then I visit coreapi repositories. Top message on each:
**This repository has been archived by the owner. It is now read-only.** .

Questions:
1. I think coreapi need active development
or
2. I think topic about api clients in official page should be removed or updated with working examples(maybe link to https://github.com/MikeWooster/api-client).

PS
to be clear, idea behind coreapi is great, to easly consume api with generated schema. But it is not finished, and not maintained.

**Comment by sigvef** [comment]
As far as I can tell, CoreAPI was deprecated in DRF 3.10, and the current suggested way is to use OpenAPI instead.

Deprecation announcement: https://www.django-rest-framework.org/community/3.10-announcement/#openapi-schema-generation

**Reply by pfk-beta** [reply]
There is CoreAPI format, and CoreAPI library. Agrre that OpenAPI is standard which we should follow. CoreAPI supports OpenAPI format. I'm talking about library, not about schemas.
